src/nextjournal/clerk/viewer.cljc

```diff
@@ -968,13 +968,27 @@
                (map (juxt #(list 'quote (symbol %)) #(->> % deref deref (list 'quote))))
                (extract-sync-atom-vars doc)))))
 
+(defn ^:private safely
+  [func error-message args->error-data & args]
+  (try (apply func args)
+    (catch #?(:clj Exception :cljs js/Error) e
+      (throw #?(:clj (ex-info error-message
+                              (apply args->error-data args)
+                              e)
+                :cljs e)))))
```

Comment:
this doesn't need to be higher-order, but doing so helps wrap only `apply-viewers*` and not the entire `->>`/`comp`, which localizes the error handling as much as possible. I think this is nice because then you know that the other two functions in the `->>` shouldn't eval user code